tests/private_beam_test.py

```diff
@@ -128,6 +130,96 @@ def test_transform_with_return_anonymized_enabled_returns_pcollection(self):
             # Assert
             self.assertIsInstance(transformed, pvalue.PCollection)
 
+    @patch('pipeline_dp.dp_engine.DPEngine.aggregate')
+    def test_vector_sum_calls_aggregate_with_params(self, mock_aggregate):
+        runner = fn_api_runner.FnApiRunner()
+        with beam.Pipeline(runner=runner) as pipeline:
+            # Arrange
+            pcol = pipeline | 'Create produce' >> beam.Create([[1, 2, 3],
+                                                               [4, 5, 6]])
+            budget_accountant = budget_accounting.NaiveBudgetAccountant(
+                total_epsilon=1, total_delta=0.01)
+            private_collection = (
+                pcol | 'Create private collection' >> private_beam.MakePrivate(
+                    budget_accountant=budget_accountant,
+                    privacy_id_extractor=PrivateBeamTest.privacy_id_extractor))
+
+            vector_sum_params = aggregate_params.VectorSumParams(
+                noise_kind=pipeline_dp.NoiseKind.GAUSSIAN,
+                max_partitions_contributed=2,
+                max_contributions_per_partition=3,
+                budget_weight=1,
+                vector_norm_kind=aggregate_params.NormKind.Linf,
+                vector_max_norm=5,
+                vector_size=3,
+                partition_extractor=lambda x: f"pk:{x // 10}",
+                value_extractor=lambda x: x,
+            )
+
+            # Act
+            transformer = private_beam.VectorSum(
+                vector_sum_params=vector_sum_params, public_partitions=[])
+            private_collection | transformer
+
+            # Assert
+            self.assertEqual(transformer._budget_accountant, budget_accountant)
+            mock_aggregate.assert_called_once()
+
+            args = mock_aggregate.call_args[0]
+
+            params = pipeline_dp.AggregateParams(
+                noise_kind=pipeline_dp.NoiseKind.GAUSSIAN,
+                metrics=[pipeline_dp.Metrics.VECTOR_SUM],
+                max_partitions_contributed=vector_sum_params.
+                max_partitions_contributed,
+                max_contributions_per_partition=vector_sum_params.
+                max_contributions_per_partition,
+                vector_norm_kind=aggregate_params.NormKind.Linf,
+                vector_max_norm=5,
+                vector_size=3,
+            )
+            self.assertEqual(params, args[1])
+
+    def test_vector_sum_returns_sensible_result(self):
+        with TestPipeline() as pipeline:
+            # Arrange
+            col = [(f"{u}", "pk1", [-100.0]) for u in range(30)]
```

Comment:
Nit: to use 2 dim vectors as data (to check that it works for multi-dim vectors)